pkg/envoy/proxy.go

```diff
@@ -179,6 +179,13 @@ func (e *envoy) Run(config interface{}, epoch int, abort <-chan error) error {
 
 	/* #nosec */
 	cmd := exec.Command(e.Config.BinaryPath, args...)
+
+	if e.Config.ProxyMetadata != nil {
```

Comment:
Is there a better mechanism that using Env here? 

Possibly putting it in the bootstrap json? 

https://github.com/istio/istio/blob/07782fd8d2ff0225af494b172dc459e5c9ace04a/pkg/bootstrap/instance.go#L35

Reply:
I am a bit confused what we are trying to do here? `ProxyMetadata` is really about setting env vars for `istio-agent` to access (for feature flags, and anything `ISTIO_META_` is put in the node bootstrap, but it looks like this is setting it for the envoy process, which shouldn't really do anything?

Reply:
The workload command sets a lot of metadata in this file and I don't see how we'd ever send that along with discovery requests to istiod. Seems like we load cluster.env/sidecar.env by using env vars – but maybe we need to do this earlier and any vars in agent need to also check values form this file?

Or should we not be setting metadata via proxyconfig if we expect it to be used? Do we expect to load this into the env elsewhere?

Reply:
Don't we read the cluster.env: https://github.com/istio/istio/blob/e8abb2e69381a8e52d06ae720075d033b190bfd7/tools/packaging/common/istio-start.sh#L37

then later run the agent: https://github.com/istio/istio/blob/e8abb2e69381a8e52d06ae720075d033b190bfd7/tools/packaging/common/istio-start.sh#L124

?

Reply:
Oh but we never read ProxyMetadata, I see

Reply:
Yeah its pretty broken, this is part of why we are moving Gateways to be injected, they suffer the same issue.

Its tricky to change the code to read ProxyMetadata instead of env since a lot of them are spaghetti global variables that are read immediately at initialization, before we could possibly read the ProxyMetadata file (so os.SetEnv in the agent isn't going to work right)

Reply:
So for #29125 we need to move pretty much everything into cluster.env or somehow read this before agent init.. 

Or read this file and have a bunch of conditionals for all the global vars...

Reply:
Alright so I updated this to include the proxyconfig.metadata in node metadata. It might not be possible to set feature flags via proxy config, but it feels like setting metadata (network/autoregistration flag) should be allowed.

Reply:
I'm also going to use this PR to convert our VM tests to use `istioctl workload ...`